### ISSUE TITLE
fix remaining 0.03% of failing test cases from LLVM test case suite

### ISSUE THREAD

**Issue by mewmew** [llir/llvm#111]
Currently we have integrated test cases from Coreutils, SQLite and the official LLVM project, and as of #110 we are passing all test cases of implemented language features of LLVM. There are still a few language features left to implement, that were added in LLVM 9.0, and these are tracked by #101, #106, and #43.

To reduce the time needed to run `go test` on `llir/llvm`, we have only enabled a carefully selected subset of the test cases, as there are 20k+ test cases of LLVM.

As of rev fba5559079e076debfe70cb2bbb81bf183f28437, if we were to enable all test cases of [Coreutils](https://github.com/llir/testdata/tree/master/coreutils), [SQLite](https://github.com/llir/testdata/tree/master/sqlite) and [LLVM](https://github.com/llir/testdata/tree/master/llvm), we would pass ~96.5% of the test cases (20010/20723 = 96.5%).

As such, this issue tracks the remaining 3.5% of the test cases which are currently failing (713 test cases fail of the total 20723 test cases).

We will start to categorize the failing test cases, and quite likely there are some ~20 language concepts left to implement which will help us pass all test cases of LLVM.

The list of currently failing test cases are as follows.

# Not yet resolved (0.2%)

## Hexadecimal floating-point literals (0.2% of failing test cases)

0.2% of currently failing test cases (2/713) are related to hexadecimal floating-point literals.

This is tracked by issue #30.

<details><summary>remaining test cases failing</summary><br>

* [llvm/test/CodeGen/PowerPC/fast-isel-call.ll](https://github.com/llir/testdata/blob/68fdb7c8ce371954493b215f547f5261572aadae/llvm/test/CodeGen/PowerPC/fast-isel-call.ll)
    - strconv.ParseUint: parsing "1397723CCABD0000401C666660000000`value out of range
* [llvm/test/Transforms/InstCombine/fneg.ll](https://github.com/llir/testdata/blob/68fdb7c8ce371954493b215f547f5261572aadae/llvm/test/Transforms/InstCombine/fneg.ll)
    - strconv.ParseUint: parsing "7FF80000000000000`value out of range

</details>

**Comment by mewmew** [comment]
@dannypsnl, would you like to take a look at the missing enums? :)

They are as follows:

* [llvm/test/Bitcode/DISubprogram-v5.ll](https://github.com/llir/testdata/blob/68fdb7c8ce371954493b215f547f5261572aadae/llvm/test/Bitcode/DISubprogram-v5.ll)
	- unable to locate DIFlag enum corresponding to "DIFlagMainSubprogram"
* [llvm/test/Bitcode/upgrade-dbg-checksum.ll](https://github.com/llir/testdata/blob/68fdb7c8ce371954493b215f547f5261572aadae/llvm/test/Bitcode/upgrade-dbg-checksum.ll)
	- unable to locate ChecksumKind enum corresponding to "CSK_None"
* [llvm/test/CodeGen/AArch64/shadow-call-stack.ll](https://github.com/llir/testdata/blob/68fdb7c8ce371954493b215f547f5261572aadae/llvm/test/CodeGen/AArch64/shadow-call-stack.ll)
	- unable to locate FuncAttr enum corresponding to "shadowcallstack"
* [llvm/test/CodeGen/X86/nocf_check.ll](https://github.com/llir/testdata/blob/68fdb7c8ce371954493b215f547f5261572aadae/llvm/test/CodeGen/X86/nocf_check.ll)
	- unable to locate FuncAttr enum corresponding to "nocf_check"
* [llvm/test/DebugInfo/Generic/directives-only.ll](https://github.com/llir/testdata/blob/68fdb7c8ce371954493b215f547f5261572aadae/llvm/test/DebugInfo/Generic/directives-only.ll)
	- unable to locate EmissionKind enum corresponding to "DebugDirectivesOnly"
* [llvm/test/DebugInfo/NVPTX/debug-empty.ll](https://github.com/llir/testdata/blob/68fdb7c8ce371954493b215f547f5261572aadae/llvm/test/DebugInfo/NVPTX/debug-empty.ll)
	- unable to locate EmissionKind enum corresponding to "DebugDirectivesOnly"
* [llvm/test/DebugInfo/NVPTX/debug-file-loc-only.ll](https://github.com/llir/testdata/blob/68fdb7c8ce371954493b215f547f5261572aadae/llvm/test/DebugInfo/NVPTX/debug-file-loc-only.ll)
	- unable to locate EmissionKind enum corresponding to "DebugDirectivesOnly"
* [llvm/test/Transforms/SimplifyCFG/opt-for-fuzzing.ll](https://github.com/llir/testdata/blob/68fdb7c8ce371954493b215f547f5261572aadae/llvm/test/Transforms/SimplifyCFG/opt-for-fuzzing.ll)
	- unable to locate FuncAttr enum corresponding to "optforfuzzing"

**Comment by mewmew** [comment]
# Resolved (34%)

## Alignment attribute of function definitions (34% of previously failing test cases)

34% of the previously failing test cases (249/713) were related to alignment attribute of function definitions.

Fixed in rev e68d900f2a6024f24b61c5e96420f6ba1cc8c37b.

<details><summary>test cases failing related to identifiers</summary><br>

* `llvm/test/Analysis/CostModel/SystemZ/intrinsic-cost-crash.ll`
* `llvm/test/Analysis/DependenceAnalysis/MIVCheckConst.ll`
* `llvm/test/Analysis/DependenceAnalysis/NonAffineExpr.ll`
* `llvm/test/Analysis/MemorySSA/cyclicphi.ll`
* `llvm/test/Analysis/MemorySSA/loop-rotate-inv-template.ll`
* `llvm/test/Analysis/MemorySSA/loop_rotate_remove_trivial_phis.ll`
* `llvm/test/Analysis/MemorySSA/reduce_clobber_limit.ll`
* `llvm/test/Analysis/MemorySSA/update_unroll.ll`
* `llvm/test/Analysis/ScalarEvolution/pr22674.ll`
* `llvm/test/Analysis/TypeBasedAliasAnalysis/PR17620.ll`
* `llvm/test/Analysis/ValueTracking/func-ptr-lsb.ll`
* `llvm/test/Bitcode/compatibility-3.8.ll`
* `llvm/test/Bitcode/compatibility-3.9.ll`
* `llvm/test/Bitcode/compatibility-4.0.ll`
* `llvm/test/Bitcode/compatibility-5.0.ll`
* `llvm/test/Bitcode/compatibility-6.0.ll`
* `llvm/test/CodeGen/AArch64/PBQP-csr.ll`
* `llvm/test/CodeGen/AArch64/analyzecmp.ll`
* `llvm/test/CodeGen/AArch64/and-mask-removal.ll`
* `llvm/test/CodeGen/AArch64/arm64-2011-03-09-CPSRSpill.ll`
* `llvm/test/CodeGen/AArch64/arm64-2012-01-11-ComparisonDAGCrash.ll`
* `llvm/test/CodeGen/AArch64/arm64-andCmpBrToTBZ.ll`
* `llvm/test/CodeGen/AArch64/arm64-atomic.ll`
* `llvm/test/CodeGen/AArch64/arm64-bitfield-extract.ll`
* `llvm/test/CodeGen/AArch64/branch-relax-alignment.ll`
* `llvm/test/CodeGen/AArch64/sched-past-vector-ldst.ll`
* `llvm/test/CodeGen/AArch64/vector_merge_dep_check.ll`
* `llvm/test/CodeGen/AMDGPU/hsa-func-align.ll`
* `llvm/test/CodeGen/AMDGPU/hsa-func.ll`
* `llvm/test/CodeGen/AMDGPU/nop-data.ll`
* `llvm/test/CodeGen/ARM/2010-06-21-nondarwin-tc.ll`
* `llvm/test/CodeGen/ARM/2010-07-26-GlobalMerge.ll`
* `llvm/test/CodeGen/ARM/2010-08-04-StackVariable.ll`
* `llvm/test/CodeGen/ARM/2011-05-04-MultipleLandingPadSuccs.ll`
* `llvm/test/CodeGen/ARM/2012-01-24-RegSequenceLiveRange.ll`
* `llvm/test/CodeGen/ARM/2012-01-26-CoalescerBug.ll`
* `llvm/test/CodeGen/ARM/2012-04-02-TwoAddrInstrCrash.ll`
* `llvm/test/CodeGen/ARM/2012-04-10-DAGCombine.ll`
* `llvm/test/CodeGen/ARM/2016-05-01-RegScavengerAssert.ll`
* `llvm/test/CodeGen/ARM/Windows/wineh-basic.ll`
* `llvm/test/CodeGen/ARM/alias_store.ll`
* `llvm/test/CodeGen/ARM/arm-and-tst-peephole.ll`
* `llvm/test/CodeGen/ARM/call-tc.ll`
* `llvm/test/CodeGen/ARM/crash-greedy.ll`
* `llvm/test/CodeGen/ARM/crash.ll`
* `llvm/test/CodeGen/ARM/fast-isel-align.ll`
* `llvm/test/CodeGen/ARM/gv-stubs-crash.ll`
* `llvm/test/CodeGen/ARM/ifcvt-branch-weight-bug.ll`
* `llvm/test/CodeGen/ARM/inlineasm-switch-mode-oneway-from-arm.ll`
* `llvm/test/CodeGen/ARM/inlineasm-switch-mode-oneway-from-thumb.ll`
* `llvm/test/CodeGen/ARM/inlineasm-switch-mode.ll`
* `llvm/test/CodeGen/ARM/interval-update-remat.ll`
* `llvm/test/CodeGen/ARM/ldrcppic.ll`
* `llvm/test/CodeGen/ARM/neon_spill.ll`
* `llvm/test/CodeGen/ARM/select_xform.ll`
* `llvm/test/CodeGen/ARM/vmul.ll`
* `llvm/test/CodeGen/ARM/vst2.ll`
* `llvm/test/CodeGen/Generic/exception-handling.ll`
* `llvm/test/CodeGen/Hexagon/P08214.ll`
* `llvm/test/CodeGen/Hexagon/addrmode-keepdeadphis.ll`
* `llvm/test/CodeGen/Hexagon/bit-bitsplit-regclass.ll`
* `llvm/test/CodeGen/Hexagon/bit-loop-rc-mismatch.ll`
* `llvm/test/CodeGen/Hexagon/branchfolder-keep-impdef.ll`
* `llvm/test/CodeGen/Hexagon/bug14859-iv-cleanup-lpad.ll`
* `llvm/test/CodeGen/Hexagon/bug14859-split-const-block-addr.ll`
* `llvm/test/CodeGen/Hexagon/find-loop-instr.ll`
* `llvm/test/CodeGen/Hexagon/global-ctor-pcrel.ll`
* `llvm/test/CodeGen/Hexagon/hexagon-verify-implicit-use.ll`
* `llvm/test/CodeGen/Hexagon/hvx-loopidiom-memcpy.ll`
* `llvm/test/CodeGen/Hexagon/hwloop-subreg.ll`
* `llvm/test/CodeGen/Hexagon/opt-addr-mode-subreg-use.ll`
* `llvm/test/CodeGen/Hexagon/rdf-filter-defs.ll`
* `llvm/test/CodeGen/Hexagon/regscavengerbug.ll`
* `llvm/test/CodeGen/Hexagon/save-kill-csr.ll`
* `llvm/test/CodeGen/Hexagon/swp-change-deps.ll`
* `llvm/test/CodeGen/Hexagon/swp-loop-carried-crash.ll`
* `llvm/test/CodeGen/Mips/pbqp-reserved-physreg.ll`
* `llvm/test/CodeGen/NVPTX/function-align.ll`
* `llvm/test/CodeGen/PowerPC/BreakableToken-reduced.ll`
* `llvm/test/CodeGen/PowerPC/aantidep-inline-asm-use.ll`
* `llvm/test/CodeGen/PowerPC/ec-input.ll`
* `llvm/test/CodeGen/PowerPC/fast-isel-icmp-split.ll`
* `llvm/test/CodeGen/PowerPC/fast-isel-load-store-vsx.ll`
* `llvm/test/CodeGen/PowerPC/fastcc_stacksize.ll`
* `llvm/test/CodeGen/PowerPC/glob-comp-aa-crash.ll`
* `llvm/test/CodeGen/PowerPC/licm-remat.ll`
* `llvm/test/CodeGen/PowerPC/memset-nc-le.ll`
* `llvm/test/CodeGen/PowerPC/memset-nc.ll`
* `llvm/test/CodeGen/PowerPC/merge-st-chain-op.ll`
* `llvm/test/CodeGen/PowerPC/pr15031.ll`
* `llvm/test/CodeGen/PowerPC/pr18663-2.ll`
* `llvm/test/CodeGen/PowerPC/pr18663.ll`
* `llvm/test/CodeGen/PowerPC/pr24636.ll`
* `llvm/test/CodeGen/PowerPC/pr27350.ll`
* `llvm/test/CodeGen/PowerPC/pr28130.ll`
* `llvm/test/CodeGen/PowerPC/rlwimi-and.ll`
* `llvm/test/CodeGen/PowerPC/split-index-tc.ll`
* `llvm/test/CodeGen/PowerPC/stwu8.ll`
* `llvm/test/CodeGen/PowerPC/tls-cse.ll`
* `llvm/test/CodeGen/PowerPC/toc-load-sched-bug.ll`
* `llvm/test/CodeGen/SPARC/missinglabel.ll`
* `llvm/test/CodeGen/Thumb/2011-05-11-DAGLegalizer.ll`
* `llvm/test/CodeGen/Thumb2/constant-islands.ll`
* `llvm/test/CodeGen/Thumb2/crash.ll`
* `llvm/test/CodeGen/Thumb2/ifcvt-rescan-bug-2016-08-22.ll`
* `llvm/test/CodeGen/X86/2009-11-17-UpdateTerminator.ll`
* `llvm/test/CodeGen/X86/2010-01-13-OptExtBug.ll`
* `llvm/test/CodeGen/X86/2010-03-17-ISelBug.ll`
* `llvm/test/CodeGen/X86/2010-04-06-SSEDomainFixCrash.ll`
* `llvm/test/CodeGen/X86/2010-04-29-CoalescerCrash.ll`
* `llvm/test/CodeGen/X86/2010-06-01-DeadArg-DbgInfo.ll`
* `llvm/test/CodeGen/X86/2010-07-11-FPStackLoneUse.ll`
* `llvm/test/CodeGen/X86/2010-08-04-StackVariable.ll`
* `llvm/test/CodeGen/X86/2010-09-30-CMOV-JumpTable-PHI.ll`
* `llvm/test/CodeGen/X86/2011-03-08-Sched-crash.ll`
* `llvm/test/CodeGen/X86/2012-01-10-UndefExceptionEdge.ll`
* `llvm/test/CodeGen/X86/2012-04-09-TwoAddrPassBug.ll`
* `llvm/test/CodeGen/X86/2012-05-19-CoalescerCrash.ll`
* `llvm/test/CodeGen/X86/2012-08-28-UnsafeMathCrash.ll`
* `llvm/test/CodeGen/X86/2012-10-02-DAGCycle.ll`
* `llvm/test/CodeGen/X86/2012-10-03-DAGCycle.ll`
* `llvm/test/CodeGen/X86/2012-12-14-v8fp80-crash.ll`
* `llvm/test/CodeGen/X86/asm-label2.ll`
* `llvm/test/CodeGen/X86/atom-fixup-lea4.ll`
* `llvm/test/CodeGen/X86/cmp.ll`
* `llvm/test/CodeGen/X86/code_placement_eh.ll`
* `llvm/test/CodeGen/X86/crash.ll`
* `llvm/test/CodeGen/X86/dbg-changes-codegen.ll`
* `llvm/test/CodeGen/X86/debug-loclists.ll`
* `llvm/test/CodeGen/X86/fast-isel-x86-64.ll`
* `llvm/test/CodeGen/X86/hidden-vis-pic.ll`
* `llvm/test/CodeGen/X86/large-gep-chain.ll`
* `llvm/test/CodeGen/X86/merge-vector-stores-scale-idx-crash.ll`
* `llvm/test/CodeGen/X86/misched-aa-colored.ll`
* `llvm/test/CodeGen/X86/misched-aa-mmos.ll`
* `llvm/test/CodeGen/X86/misched-new.ll`
* `llvm/test/CodeGen/X86/norex-subreg.ll`
* `llvm/test/CodeGen/X86/pr14098.ll`
* `llvm/test/CodeGen/X86/pr38743.ll`
* `llvm/test/CodeGen/X86/stdcall-notailcall.ll`
* `llvm/test/CodeGen/X86/tail-dup-repeat.ll`
* `llvm/test/CodeGen/X86/tailcall-returndup-void.ll`
* `llvm/test/CodeGen/X86/tailcall-ri64.ll`
* `llvm/test/CodeGen/X86/twoaddr-sink-terminator.ll`
* `llvm/test/CodeGen/X86/vector-trunc-widen.ll`
* `llvm/test/CodeGen/X86/x86-mixed-alignment-dagcombine.ll`
* `llvm/test/DebugInfo/AArch64/cfi-eof-prologue.ll`
* `llvm/test/DebugInfo/ARM/cfi-eof-prologue.ll`
* `llvm/test/DebugInfo/COFF/fpo-argsize.ll`
* `llvm/test/DebugInfo/COFF/lambda.ll`
* `llvm/test/DebugInfo/COFF/scopes.ll`
* `llvm/test/DebugInfo/COFF/type-quals.ll`
* `llvm/test/DebugInfo/COFF/types-calling-conv.ll`
* `llvm/test/DebugInfo/COFF/types-recursive-unnamed.ll`
* `llvm/test/DebugInfo/Generic/2010-03-24-MemberFn.ll`
* `llvm/test/DebugInfo/Generic/2010-04-06-NestedFnDbgInfo.ll`
* `llvm/test/DebugInfo/Generic/PR20038.ll`
* `llvm/test/DebugInfo/Generic/def-line.ll`
* `llvm/test/DebugInfo/Generic/dwarf-public-names.ll`
* `llvm/test/DebugInfo/Generic/incorrect-variable-debugloc.ll`
* `llvm/test/DebugInfo/Generic/linkage-name-abstract.ll`
* `llvm/test/DebugInfo/Generic/member-order.ll`
* `llvm/test/DebugInfo/Generic/recursive_inlining.ll`
* `llvm/test/DebugInfo/Generic/tu-composite.ll`
* `llvm/test/DebugInfo/X86/2011-12-16-BadStructRef.ll`
* `llvm/test/DebugInfo/X86/DW_AT_linkage_name.ll`
* `llvm/test/DebugInfo/X86/DW_AT_object_pointer.ll`
* `llvm/test/DebugInfo/X86/cu-ranges-odr.ll`
* `llvm/test/DebugInfo/X86/dbg-declare-arg.ll`
* `llvm/test/DebugInfo/X86/dwarf-public-names.ll`
* `llvm/test/DebugInfo/X86/elf-names.ll`
* `llvm/test/DebugInfo/X86/fission-inline.ll`
* `llvm/test/DebugInfo/X86/generate-odr-hash.ll`
* `llvm/test/DebugInfo/X86/gnu-public-names.ll`
* `llvm/test/DebugInfo/X86/linkage-name.ll`
* `llvm/test/DebugInfo/X86/pr11300.ll`
* `llvm/test/DebugInfo/X86/pr12831.ll`
* `llvm/test/DebugInfo/X86/reference-argument.ll`
* `llvm/test/DebugInfo/X86/sret.ll`
* `llvm/test/DebugInfo/X86/void-typedef.ll`
* `llvm/test/DebugInfo/X86/xray-split-dwarf-interaction.ll`
* `llvm/test/ExecutionEngine/OrcLazy/global-ctors-and-dtors.ll`
* `llvm/test/ExecutionEngine/OrcLazy/hello.ll`
* `llvm/test/Feature/alignment.ll`
* `llvm/test/Instrumentation/SanitizerCoverage/coverage-dbg.ll`
* `llvm/test/LTO/Resolution/X86/Inputs/appending-var-2.ll`
* `llvm/test/Linker/Inputs/alignment.ll`
* `llvm/test/Linker/Inputs/replaced-function-matches-first-subprogram.ll`
* `llvm/test/Linker/alignment.ll`
* `llvm/test/Linker/replaced-function-matches-first-subprogram.ll`
* `llvm/test/Linker/type-unique-odr-b.ll`
* `llvm/test/Linker/type-unique-simple2-a.ll`
* `llvm/test/Linker/type-unique-simple2-b.ll`
* `llvm/test/Linker/type-unique-type-array-a.ll`
* `llvm/test/Linker/type-unique-type-array-b.ll`
* `llvm/test/SymbolRewriter/rewrite.ll`
* `llvm/test/ThinLTO/X86/Inputs/dicompositetype-unique-alias.ll`
* `llvm/test/ThinLTO/X86/Inputs/dicompositetype-unique2.ll`
* `llvm/test/ThinLTO/X86/Inputs/referenced_by_constant.ll`
* `llvm/test/ThinLTO/X86/dicompositetype-unique-alias.ll`
* `llvm/test/ThinLTO/X86/dicompositetype-unique2.ll`
* `llvm/test/Transforms/AlignmentFromAssumptions/start-unk.ll`
* `llvm/test/Transforms/CodeGenPrepare/basic.ll`
* `llvm/test/Transforms/DeadStoreElimination/cs-cs-aliasing.ll`
* `llvm/test/Transforms/GVN/crash.ll`
* `llvm/test/Transforms/GVN/nonescaping-malloc.ll`
* `llvm/test/Transforms/GVN/stale-loop-info.ll`
* `llvm/test/Transforms/GlobalDCE/global_ctors_integration.ll`
* `llvm/test/Transforms/GlobalOpt/cxx-dtor.ll`
* `llvm/test/Transforms/Inline/crash2.ll`
* `llvm/test/Transforms/Inline/devirtualize-2.ll`
* `llvm/test/Transforms/Inline/devirtualize.ll`
* `llvm/test/Transforms/InstCombine/crash.ll`
* `llvm/test/Transforms/InstSimplify/require-dominator.ll`
* `llvm/test/Transforms/JumpThreading/ddt-crash.ll`
* `llvm/test/Transforms/JumpThreading/ddt-crash2.ll`
* `llvm/test/Transforms/JumpThreading/landing-pad.ll`
* `llvm/test/Transforms/LICM/loopsink-pr39570.ll`
* `llvm/test/Transforms/LoopDeletion/2011-06-21-phioperands.ll`
* `llvm/test/Transforms/LoopSimplify/dbg-loc.ll`
* `llvm/test/Transforms/LoopStrengthReduce/2011-10-14-IntPtr.ll`
* `llvm/test/Transforms/LoopStrengthReduce/2012-03-15-nopreheader.ll`
* `llvm/test/Transforms/LoopStrengthReduce/2013-01-14-ReuseCast.ll`
* `llvm/test/Transforms/LoopStrengthReduce/X86/2009-11-10-LSRCrash.ll`
* `llvm/test/Transforms/LoopUnroll/2011-08-08-PhiUpdate.ll`
* `llvm/test/Transforms/LoopUnroll/2011-08-09-PhiUpdate.ll`
* `llvm/test/Transforms/LoopUnswitch/2011-09-26-EHCrash.ll`
* `llvm/test/Transforms/LoopUnswitch/2012-04-30-LoopUnswitch-LPad-Crash.ll`
* `llvm/test/Transforms/LoopVectorize/X86/vector_ptr_load_store.ll`
* `llvm/test/Transforms/LoopVectorize/lcssa-crash.ll`
* `llvm/test/Transforms/MergeFunc/2011-02-08-RemoveEqual.ll`
* `llvm/test/Transforms/MergeFunc/2013-01-10-MergeFuncAssert.ll`
* `llvm/test/Transforms/MergeFunc/crash.ll`
* `llvm/test/Transforms/MergeFunc/inttoptr-address-space.ll`
* `llvm/test/Transforms/MergeFunc/inttoptr.ll`
* `llvm/test/Transforms/MergeFunc/vector.ll`
* `llvm/test/Transforms/NewGVN/crash.ll`
* `llvm/test/Transforms/NewGVN/nonescaping-malloc.ll`
* `llvm/test/Transforms/NewGVN/phi-edge-handling.ll`
* `llvm/test/Transforms/NewGVN/pr31501.ll`
* `llvm/test/Transforms/NewGVN/stale-loop-info.ll`
* `llvm/test/Transforms/PhaseOrdering/2010-03-22-empty-baseclass.ll`
* `llvm/test/Transforms/PhaseOrdering/gdce.ll`
* `llvm/test/Transforms/Reg2Mem/crash.ll`
* `llvm/test/Transforms/SLPVectorizer/X86/crash_bullet3.ll`
* `llvm/test/Transforms/SLPVectorizer/X86/resched.ll`
* `llvm/test/Transforms/SimpleLoopUnswitch/2011-09-26-EHCrash.ll`
* `llvm/test/Transforms/SimpleLoopUnswitch/2012-04-30-LoopUnswitch-LPad-Crash.ll`
* `llvm/test/Transforms/Util/strip-nonlinetable-debuginfo-containingtypes.ll`

</details>

**Comment by mewmew** [comment]
To prioritize for the v0.3 release, these test cases have been targeted for a future release of llir/llvm.

## Update to grammar needed (0.8% of failing test cases)

0.8% of currently failing test cases (6/713) are related to grammar.

### Grammar likely related to Function String Attribute

ref: #40.

<details><summary>test cases failing likely related to Function String Attribute grammar</summary><br>

* `llvm/test/Bitcode/attributes.ll`
	- syntax error at [line 266](https://github.com/llir/testdata/blob/68fdb7c8ce371954493b215f547f5261572aadae/llvm/test/Bitcode/attributes.ll#L266)
* `llvm/test/Transforms/Inline/inline-varargs.ll`
	- syntax error at [line 6](https://github.com/llir/testdata/blob/68fdb7c8ce371954493b215f547f5261572aadae/llvm/test/Transforms/Inline/inline-varargs.ll#L6)

</details>

### Grammar related to Module Summary

ref: #43.

<details><summary>test cases failing related to Module Summary grammar</summary><br>

* `llvm/test/Assembler/thinlto-summary.ll`
	- syntax error at [line 6](https://github.com/llir/testdata/blob/68fdb7c8ce371954493b215f547f5261572aadae/llvm/test/Assembler/thinlto-summary.ll#L6)
* `llvm/test/Assembler/thinlto-vtable-summary.ll`
	- syntax error at [line 30](https://github.com/llir/testdata/blob/68fdb7c8ce371954493b215f547f5261572aadae/llvm/test/Assembler/thinlto-vtable-summary.ll#L30)
* `llvm/test/LTO/X86/Inputs/dllimport.ll`
	- syntax error at [line 16](https://github.com/llir/testdata/blob/68fdb7c8ce371954493b215f547f5261572aadae/llvm/test/LTO/X86/Inputs/dllimport.ll#L16)
* `llvm/test/LTO/X86/dllimport.ll`
	- syntax error at [line 28](https://github.com/llir/testdata/blob/68fdb7c8ce371954493b215f547f5261572aadae/llvm/test/LTO/X86/dllimport.ll#L28)

</details>

**Comment by dannypsnl** [comment]
For literal `1397723CCABD0000401C666660000000`, `u0x400000000` and `7FF80000000000000` I created a quick reproduce example:

```go
func main() {
	if _, err := strconv.ParseUint("1397723CCABD0000401C666660000000", 16, 64); err != nil {
		fmt.Printf("%s\n", err)
	}
	if _, err := strconv.ParseUint("u0x400000000", 2, 64); err != nil {
		fmt.Printf("%s\n", err)
	}
	if _, err := strconv.ParseUint("7FF80000000000000", 16, 64); err != nil {
		fmt.Printf("%s\n", err)
	}
}
// Output:
//
// strconv.ParseUint: parsing "1397723CCABD0000401C666660000000": value out of range
// strconv.ParseUint: parsing "u0x400000000": invalid syntax
// strconv.ParseUint: parsing "7FF80000000000000": value out of range
```

For out of range we probably need something like https://github.com/cockroachdb/cockroach/blob/master/pkg/util/uint128/uint128.go

That's the easy part, but for `u0x400000000` I can't find document mentions how LLVM integer literal express. Seems like I have to take a look at IR Tokenizer.

**Comment by mewmew** [comment]
@dannypsnl, thanks for taking a look at this! Here are all the "value out of range" errors I get when parsing all test cases:

* `Analysis/CostModel/X86/gep.ll`
	- unable to parse integer "295147905179352825855"; strconv.ParseInt: parsing "295147905179352825855": value out of range
* `Transforms/InstCombine/fneg.ll`
	- strconv.ParseUint: parsing "7FF80000000000000": value out of range
* `Transforms/SCCP/apint-bigint2.ll`
	- unable to parse integer "9224497936761618431"; strconv.ParseInt: parsing "9224497936761618431": value out of range
* `CodeGen/X86/huge-stack-offset2.ll`
	- unable to parse unsigned integer literal "u0x400000000"; strconv.ParseUint: parsing "u0x400000000": invalid syntax
* `CodeGen/X86/getelementptr.ll`
	- unable to parse integer "39614081257132168796771975167"; strconv.ParseInt: parsing "39614081257132168796771975167": value out of range
* `CodeGen/PowerPC/fast-isel-call.ll`
	- strconv.ParseUint: parsing "1397723CCABD0000401C666660000000": value out of range

**Comment by mewmew** [comment]
> That's the easy part, but for u0x400000000 I can't find document mentions how LLVM integer literal express. Seems like I have to take a look at IR Tokenizer.

Signed and unsigned hexadecimal integers. Here's another example (from `llvm/test/Integer/constexpr_bt.ll`):

* signed hexadecimal integers `s0x012312`

**Comment by mewmew** [comment]
> For literal 1397723CCABD0000401C666660000000, u0x400000000 and 7FF80000000000000 I created a quick reproduce example:

I took a quick look at the test case containing `1397723CCABD0000401C666660000000` (`llvm/test/CodeGen/PowerPC/fast-isel-call.ll`) and it turns out its a floating-point constant in hexadecimal notation. I'll make sure to update the error message so it doesn't strip the `0x` prefix.

As it's a floating-point constant and not an integer constant, this test case would be fixed by #30.

**Comment by mewmew** [comment]
All test cases are now resolved, except for those related to hexadecimal floating-pointer literals. There is a dedicated issue to track progress on adding complete support for hexadecimal floating-point literals, namely #30.

A few test cases are related to module summaries, and a few other aspects of LLVM IR that have been marked for a future release. The dedicated tracking issues are #40, #43, #117 and #118.

As such, we consider this meta issue complete and will it close it for now. Follow-up for future discussions are referred to the issue threads of each dedicated issue.
